### PR TITLE
#122 GitHub CI

### LINKED ISSUE #122: Use Github for CI

### PR DESCRIPTION
**Pull request by driccio**
Add a workflow to manage CI (via github actions)
On each push this workflow will be launched. If one step fails, then author of the commit receive an email.
You can see the result here : https://github.com/4sh/datamaintain/actions/runs/721968944

Resolves #122

Reviewers : @Lysoun @nroulon

### CODE REVIEW

File: .github/workflows/CI.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      - name: Upload Unit Test Results
+        if: always()
+        uses: actions/upload-artifact@v2
+        with:
+          name: tests-results # Name artifact for storage in cache
+          path: |
+            modules/cli/build/test-results/**/*.xml
+            modules/core/build/test-results/**/*.xml
+            modules/driver-mongo/build/test-results/**/*.xml
+            modules/test/build/test-results/**/*.xml
```
Comment (Lysoun) [inline .github/workflows/CI.yaml]: I am afraid that tests results for new modules will be forgotten if paths are declared like that. Wouldn't that be possible to declare a unique path like that: ```modules/**/build/test-results/**/*.xml```

**Review by Lysoun — COMMENTED** [review]
I am very new to this but isn't there a way to display the test results that were uploaded on github? Right know to display the detail, I seem to have to download them, which is not very user-friendly. 

Could you try to break a test on purpose and commit with the broken test, just to see what is shown in github to help the developer acknowledge the problem, find it and fix it?

**Comment by driccio** [comment]
Hello @Lysoun, thanks for the review. 

I made a try with failing tests. You can see job [here](https://github.com/4sh/datamaintain/runs/2497927287)

The rendering is not perfect. But imho it does the job. What do you think ?

**Comment by Lysoun** [comment]
@driccio It looks good to me!

**Pull request merged**